compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaUnionTypeSymbol.java

```diff
@@ -97,4 +115,18 @@ public String signature() {
             return joiner.toString();
         }
     }
+
+    private String getSignatureForFiniteType() {
+        List<TypeSymbol> memberTypes = this.memberTypeDescriptors();
+        if (memberTypes.size() == 2 &&
+                memberTypes.get(1).typeKind() == TypeDescKind.NIL) {
```

Comment:
is it possible to have NIL in the first element?

Reply:
Ah no.. Can't really assume that. Was wondering whether to get rid of this altogether for finite types :thinking:

Reply:
Even for unions, should we always print unions which contain `()` as `T1|T2|T3?`?

Reply:
That is a readability issue. For optional types saying `T?` is more natural than saying `T|()` . But When we have more I find `T1|T2|T3|()` is more natural than `T1|T2|T3?`

Reply:
Yeah, true. Removed this for finite types and fixed the optional type logic for unions and updated.